sysl2/sysl/seqs.go

```diff
@@ -0,0 +1,240 @@
+package main
+
+import (
+	"flag"
+	"io"
+	"sort"
+	"strings"
+
+	"github.com/anz-bank/sysl/src/proto"
+	"github.com/anz-bank/sysl/sysl2/sysl/seqs"
+	log "github.com/sirupsen/logrus"
+)
+
+type SimpleParser struct {
+	self string
+}
+
+type sequenceDiagParam struct {
+	seqs.AppLabeler
+	seqs.EndpointLabeler
+	endpoints  []string
+	title      string
+	blackboxes [][]string
+}
+
+func generateSequenceDiag(m *sysl.Module, p *sequenceDiagParam) (string, error) {
+	w := seqs.MakeSequenceDiagramWriter(true, "skinparam maxMessageSize 250")
+
+	v := seqs.MakeSequenceDiagramVisitor(p.AppLabeler, p.EndpointLabeler, w, m)
+
+	e := seqs.MakeEndpointCollectionElement(p.title, p.endpoints, p.blackboxes)
+
+	e.Accept(v)
+
+	return w.String(), nil
+}
+
+type arrayFlags []string
+
+// implement the String method of Value interface in flag.go
+func (i *arrayFlags) String() string {
+	return strings.Join(*i, ",")
+}
+
+// implement the Set method of Value interface in flag.go
+func (i *arrayFlags) Set(value string) error {
+	*i = append(*i, value)
+	return nil
+}
+
+func loadApp(root string, models []string) *sysl.Module {
+	// Model we want to generate seqs for
+	var model string
+	for _, val := range models {
+		model = val
+		break
+	}
```

Comment:
This looks like `model := models[0]` (empty string isn't a valid model).

Reply:
Updated to find the first non-empty string.

Reply:
models come from the command-line args. we just validate it and select the first model.

Reply:
Changed to handle one model here, args do not support multiple models now.